float8_experimental/float8_linear_utils.py

```diff
@@ -30,35 +30,52 @@ class LinearType(Enum):
     DYNAMIC = auto()
 
 
-REQUIRES_SYNC = {LinearType.DELAYED}
-
-
 def get_float8_linear(
     linear_type: LinearType,
     linear_ref: torch.nn.Linear,
     emulate: bool = False,
+    scaling_type_x: TensorScalingType = TensorScalingType.DELAYED,
+    scaling_type_w: TensorScalingType = TensorScalingType.DELAYED,
+    scaling_type_dL_dY: TensorScalingType = TensorScalingType.DELAYED,
 ):
     """Returns a Float8Linear module of the given type, initialized from linear_ref.
     Args:
         linear_type: The type of Float8Linear to return.
         linear_ref: The linear module to initialize from.
         emulate: Whether to emulate the fp8 matmul logic in float32.
+        scaling_type_x: delayed vs dynamic scaling for `x`.
+        scaling_type_w: delayed vs dynamic scaling for `w`.
+        scaling_type_dL_dY: delayed vs dynamic scaling for `dL_dY`.
     """
-    LINEAR_TYPE_MAP = {
-        LinearType.DELAYED: Float8Linear,
-        LinearType.DYNAMIC: Float8DynamicLinear,
-    }
-    if linear_type not in LINEAR_TYPE_MAP:
-        raise ValueError(f"linear_type must be one of {LINEAR_TYPE_MAP.keys()}")
-    return LINEAR_TYPE_MAP[linear_type].from_float(
-        copy.deepcopy(linear_ref),
-        emulate=emulate,
-    )
+    if linear_type is LinearType.DYNAMIC:
+        return Float8DynamicLinear.from_float(
+            copy.deepcopy(linear_ref), emulate=emulate
+        )
+    else:
+        assert linear_type is LinearType.DELAYED
+        return Float8Linear.from_float(
+            copy.deepcopy(linear_ref),
+            emulate=emulate,
+            scaling_type_x=scaling_type_x,
+            scaling_type_w=scaling_type_w,
+            scaling_type_dL_dY=scaling_type_dL_dY,
+        )
 
 
-def linear_requires_sync(linear_type: LinearType):
+def linear_requires_sync(
+    linear_type: LinearType,
+    scaling_type_x: TensorScalingType = TensorScalingType.DELAYED,
```

Comment:
Nit: we should probably remove the defaults right?

Reply:
hmm, long term probably yes.  I'm taking the approach of tackling the "what's the default" recipe separately and not changing default behavior for now.